src/compiler/crystal/types.cr

```diff
@@ -1321,9 +1328,22 @@ module Crystal
   # but can also match other types (like ints and floats) if the literal
   # fits in those types.
   class NumberLiteralType < Type
+    # The literal associated with this type
     getter literal : NumberLiteral
+
+    # The most exact match type, or the first match otherwise
     @matched_type : Type?
 
+    # A type that signals it's an ambiguous call, because the literal
+    # can match two types (for example when 1 matches both Int8 and Int16
+    # restrictions, but there's no Int32 restriction)
+    @ambiguous_call : Type?
```

Comment:
If the call is ambiguous the value of this ivar will be a union types of the possibilities?

Reply:
No, just the first type. I can change it to be an arrray of all types, but maybe it's not necessary.